### ISSUE TITLE
🐛 [firestore-bigquery-export] Script-generated schema view interprets reference fields as strings instead of objects

### ISSUE THREAD

**Issue by conceptualben** [firebase/extensions#1685]
- Extension name: **firestore-bigquery-export**
- Extension version: **0.1.33**

For what I've seen, when creating a schema view from the provided schema view generator script, the view works if a reference field is stored plaily, i.e. as a string value with the path, e.g. "mycollection/mydocumentid", but reference fields are actually getting saved in a structured way by the extension (the object with properties _firestore and _path). Additionally, the data import script does save these reference fields as strings instead, so if we only worked with past data things would be fine with the generated schema view.

I think the extension of the scripts should either:
1) Make extension store reference fields as plain strings
2) Make the data import script store reference fields in the structured object format AND update schema view generation script to support this
3) Simply update schema view generation script to work with both plain string and structured object formats.

I have manually fixed the view for my case in a way that would work. Reading the updated query probably illustrates the problem best:

As it was generated:
```
      FIRST_VALUE(JSON_EXTRACT_SCALAR(data, '$.shop')) OVER(
        PARTITION BY document_name
        ORDER BY
          timestamp DESC
      ) AS shop,
```

With my fix:
```
      FIRST_VALUE(
        COALESCE(
          JSON_EXTRACT_SCALAR(data, '$.shop'), 
          (SELECT STRING_AGG(segments, '/') FROM  UNNEST(JSON_EXTRACT_STRING_ARRAY(data, '$.shop._path.segments')) as segments)
        )
      ) OVER(
        PARTITION BY document_name
        ORDER BY
          timestamp DESC
      ) AS shop,
```

As you can see I'm using COALESCE to return whichever method works, getting it from a string or getting it from the reference object. Just a contribution if someone else is needing to do the same.

**Comment by dreinon** [comment]
Hi @conceptualben, reference fields are supposed to be strings. There was a bug in version 0.1.33 that was unintentionally pushing objects instead of strings for reference fields to bigquery. This should be solved now with #1652 and probably going out in a new patch version 0.1.34 soon.

**Comment by conceptualben** [comment]
Ah, good! Seeing the update now. Thanks!
